Sources/BraintreeCoreSwift/BTGraphQLHTTP.swift

```diff
@@ -0,0 +1,244 @@
+    }
+
+    public override func delete(_ path: String, completion: @escaping RequestCompletion) {
+        NSException(name: exceptionName, reason: "DELETE is unsupported").raise()
+    }
+
+    public override func delete(_ path: String, parameters: NSDictionary? = nil, completion: RequestCompletion?) {
+        NSException(name: exceptionName, reason: "DELETE is unsupported").raise()
+    }
+
+    // MARK: - Internal methods
+    
+    func httpRequest(
+        method: String,
+        parameters: NSDictionary? = [:],
+        completion: RequestCompletion?
+    ) {
+        var errorUserInfo: [String: Any] = [:]
+
+        if self.baseURL.absoluteString.isEmpty || self.baseURL.absoluteString == "" {
+            errorUserInfo["method"] = method
+            errorUserInfo["parameters"] = parameters
+            let error = Self.constructError(code: .missingBaseURL, userInfo: errorUserInfo)
+            completion?(nil, nil, error)
+            return
+        }
+        
+        let authorization: String
+        switch clientAuthorization {
+        case .authorizationFingerprint(let fingerprint):
+            authorization = fingerprint
+        case .tokenizationKey(let key):
+            authorization = key
+        default:
+            authorization = "" 
+        }
+        
+        guard let components = URLComponents(string: baseURL.absoluteString) else {
+            let error = Self.constructError(
+                code: .urlStringInvalid,
+                userInfo: [NSLocalizedDescriptionKey: "The URL absolute string is malformed or invalid."]
+            )
+            completion?(nil, nil, error)
+            return
+        }
+
+        guard let urlFromComponents = components.url else {
+            let error = Self.constructError(
+                code: .urlStringInvalid,
+                userInfo: [NSLocalizedDescriptionKey: "The URL absolute string is malformed or invalid."]
+            )
+            completion?(nil, nil, error)
+            return
+        }
+
+        let headers = [
+            "User-Agent": userAgentString(),
+            "Braintree-Version": BTCoreConstants.graphQLVersion,
+            "Authorization": "Bearer \(authorization)",
+            "Content-Type": "application/json; charset=utf-8"
+        ]
+        
+        var request: URLRequest
+    
+        do {
+            let bodyData = try JSONSerialization.data(withJSONObject: parameters ?? [:])
+            request = URLRequest(url: urlFromComponents)
+            request.httpBody = bodyData
+            request.allHTTPHeaderFields = headers
+            request.httpMethod = method
+            
+            let task: URLSessionTask = session.dataTask(with: request) { [weak self] data, response, error in
+                guard let self = self else { return }
+                self.handleRequestCompletion(data: data, response: response, error: error, completion: completion)
+            }
+
+            task.resume()
```

Comment:
Nitpick
```suggestion
            }.resume()
```

Reply:
Is this the approach we want to take in the entire SDK? We currently do this across several of the HTTP clients, so should update them all if we want to do:
```
session.dataTask(with: request) { [weak self] data, response, error in
    // Do request stuff
}.resume()
```
over the current:
```
let task: URLSessionTask = session.dataTask(with: request) { [weak self] data, response, error in
    // Do request stuff
}

task.resume()
```
We should add whichever we prefer to the style guide since I could see it potentially being done both ways.

Reply:
Good call to note this in the style guide! I prefer the less-line option, as it's sometimes easy to forget that the `resume()` call is what actually kicks off the session. What do you prefer?

Reply:
Added to the style guide! Wasn't sure totally how to word it so feel free to make edits.